app/websocket_app.rb

```diff
@@ -312,6 +305,52 @@ def call(env)
 
   private
 
+  def process_daemon_connection(node_name, connection, message)
+    update_node_attributes(node_name, message)
+    Async.logger.info("#{node_name.inspect} connected")
+    processor = MessageProcessor.new(node_name, connection)
```

Comment:
At some point we should look at splitting the `MessageProcessor` for the three different types:

* Regular `CONNECTED`,
* Batch connections, and
* Stepd connections (possible the same as above)

Reply:
Yes we should.  Trello card exists to track this.